Fishing Draft PR

Technobug14 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Just a draft. Fishing soon. Highly WIP, old code being fixed up
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Fishing
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
fishing
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Technobug14 committed:
draft pr changes

draft, bad code

mrpibb4201337 commented:
Owned

Technobug14 committed:
misc changes

ghgughgughgu

Technobug14 committed:
Nearly finished

FISH FISH FISH FISH FISH

Technobug14 committed:
FISH CODE BROKE

DOOR STUCK

Technobug14 committed:
Fixes

Still broken, but less broken

AndrewL97 commented on mojave/fishing/fishing.dm:
```diff
@@ -0,0 +1,48 @@
+			if(!can_fish(user))
+				return TRUE
+			to_chat(user, "<span class='notice'>You reel in your catch.</span>")
+			getFished()
```
Missing indent after `if(do_after...`

Technobug14 committed:
Sprites working

Sprites work, only telescopic fishing rod and fish to fix and implement.

Technobug14 committed:
misc changes

changes to pathing, moved some stuff around and so on. Still need to fix the fish food items to be in-line with TG food code.

Technobug14 committed:
Working version

Everything works, some stuff still needs ironing out but this is a working prototype suitable for PRing

Hekzder commented:
Closing as #1922 was merged.
